uarray/_uarray_dispatch.cxx

```diff
@@ -187,6 +448,7 @@ std::string backend_to_domain_string(PyObject * backend) {
  */
 PyObject * clear_all_globals(PyObject * /* self */, PyObject * /* args */) {
   global_domain_map.clear();
+  thread_local_domain_map.clear();
```

Comment:
We shouldn't clear `thread_local` data in this function since it will only be called on one thread. This suggests that the cleanup belongs elsewhere (probably `set_state`).